moveit_planners/ompl/ompl_interface/src/parameterization/joint_space/constrained_planning_state_space.cpp

```diff
@@ -0,0 +1,93 @@
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Jeroen De Maeyer */
+
+#include <moveit/ompl_interface/parameterization/joint_space/constrained_planning_state_space.h>
+
+#include <ompl/base/spaces/constraint/ConstrainedStateSpace.h>
+
+const std::string ompl_interface::ConstrainedPlanningStateSpace::PARAMETERIZATION_TYPE =
+    "ConstrainedPlanningJointModel";
+
+ompl_interface::ConstrainedPlanningStateSpace::ConstrainedPlanningStateSpace(
+    const ModelBasedStateSpaceSpecification& spec)
+  : ModelBasedStateSpace(spec)
+{
+  setName(getName() + "_" + PARAMETERIZATION_TYPE);
+}
+
+double* ompl_interface::ConstrainedPlanningStateSpace::getValueAddressAtIndex(ompl::base::State* ompl_state,
+                                                                              const unsigned int index) const
+{
+  if (index >= variable_count_)
+    return nullptr;
+
+  // Developer tip: replace this with a dynamic_cast for debugging
+  return ompl_state->as<ompl_interface::ConstrainedPlanningStateSpace::StateType>()->values + index;
+}
+
+void ompl_interface::ConstrainedPlanningStateSpace::copyToRobotState(moveit::core::RobotState& robot_state,
+                                                                     const ompl::base::State* ompl_state) const
+{
+  robot_state.setJointGroupPositions(
+      spec_.joint_model_group_,
+      ompl_state->as<ompl::base::ConstrainedStateSpace::StateType>()->getState()->as<StateType>()->values);
+  robot_state.update();
+}
+
+void ompl_interface::ConstrainedPlanningStateSpace::copyToOMPLState(ompl::base::State* ompl_state,
+                                                                    const moveit::core::RobotState& robot_state) const
+{
+  robot_state.copyJointGroupPositions(
+      spec_.joint_model_group_,
+      ompl_state->as<ompl::base::ConstrainedStateSpace::StateType>()->getState()->as<StateType>()->values);
+
+  // clear any cached info (such as validity known or not)
+  ompl_state->as<ompl::base::ConstrainedStateSpace::StateType>()->getState()->as<StateType>()->clearKnownInformation();
+}
+
+void ompl_interface::ConstrainedPlanningStateSpace::copyJointToOMPLState(ompl::base::State* ompl_state,
+                                                                         const moveit::core::RobotState& robot_state,
+                                                                         const moveit::core::JointModel* joint_model,
+                                                                         int ompl_state_joint_index) const
+{
```

Comment:
Resolved based on an earlier discussion:

>Looking at the examples (the tests) I don't mind this interface nearly as much. It seems like it is this way because of the interface of ompl and the intersection with the moveit stadards (the ref to the RobotState).